runtime/jcl/common/thread.cpp

```diff
@@ -417,22 +417,32 @@ Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
 void JNICALL
 Java_java_lang_Thread_setCurrentThread(JNIEnv *env, jclass clazz, jobject thread)
 {
-	Assert_JCL_unimplemented();
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	currentThread->threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+	VM_VMAccess::inlineExitVMToJNI(currentThread);
 }
 
 /* static native Object[] extentLocalCache(); */
-jobject JNICALL
+jobjectArray JNICALL
 Java_java_lang_Thread_extentLocalCache(JNIEnv *env, jclass clazz)
 {
-	Assert_JCL_unimplemented();
-	return NULL;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	jobjectArray result = (jobjectArray)VM_VMHelpers::createLocalRef(env, currentThread->extentLocalCache);
```

Comment:
same here and below

Reply:
Similar question, assuming `issueReadBarrier()` is needed, any particular scenario require it?